src/DocumentComponent.ts

```
@@ -0,0 +1,38 @@
+export class DocumentComponent extends NodeComponent<DocumentFragment> {
+    public constructor(html?: string) {
+        super(html ? document.createRange().createContextualFragment(html) : document.createDocumentFragment());
+    }
+
+    /**
+     * Template literal tag function that accepts HTML code with components in a
+     * string literal
+     */
+    public static tag(strings: TemplateStringsArray, ...components: NodeComponent<any>[]): DocumentComponent {
+        const idPrefix = `tag-${crypto.randomUUID()}-`;
+        const doc = new DocumentComponent(strings.reduce((acc, str, index) => acc += `${str}${index < components.length ? `<slot name="${idPrefix}${index - 1}"></slot>` : ""}`, ""));
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)